float32 fsr1
float32 fsr2
float32 fsr3
float32 fsr4
int32 x
int32 y
